### PR TITLE
Remove deprecation warnings for start/end arguments and iohints

### LINKED ISSUE #20449: I/O module: bug with specifying a different start for the channel than 0
### Summary of Problem
I discovered that if a file already has contents and you create a writing channel (at least using `openwriter`, though I don't see why `file.writer` would behave any differently since `openwriter` calls it), specifying a starting location other than 0 will cause all the characters prior to your starting location to be re-interpreted as binary.

This seems incorrect - I would expect the contents of the file outside of the specified region to be untouched, enabling concurrent modifications to distinct parts of the file.

### Steps to Reproduce

**Source Code:**

```chapel
use IO;

var filename = "openwriterLimited.txt";
var writeCh = openwriter(filename);
writeCh.writeln("She said she was looking for me");
writeCh.close();

var writeChStart = openwriter(filename, start=31);
writeChStart.write("us");
writeChStart.close();
```

**Compile command:**
`chpl foo.chpl`

**Execution command:**
`./foo`

**Associated Future Test(s):**
[`test/library/standard/IO/openwriterStartBug.chpl`](https://github.com/chapel-lang/chapel/blob/main/test/library/standard/IO/openwriterStartBug.chpl) #20451 

### Configuration Information

- Output of `chpl --version`: chpl version 1.28.0 pre-release
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
CHPL_TARGET_PLATFORM: darwin
CHPL_TARGET_COMPILER: clang
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
CHPL_GMP: bundled
CHPL_HWLOC: bundled
CHPL_RE2: bundled
CHPL_LLVM: none *
CHPL_AUX_FILESYS: none
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

### LINKED ISSUE #20450: I/O module: bug with specifying a different end position for a channel than `max(int(64))`
### Summary of Problem
I discovered that if a file already has contents and you create a writing channel (at least using `openwriter`, though I don't see why `file.writer` would behave any differently since `openwriter` calls it), specifying an ending location other than `max(int(64))` (the end of the file) will cause all the characters after the end location to be dropped, truncating the file after your additions.

This seems incorrect - I would expect the contents of the file outside of the specified region to be untouched, enabling concurrent modifications to distinct parts of the file.

### Steps to Reproduce

**Source Code:**

```chapel
use IO;

var filename = "openwriterLimited.txt";
var writeCh = openwriter(filename);
writeCh.writeln("She said she was looking for me");
writeCh.close();

var writeChEnd = openwriter(filename, end=3);
writeChEnd.write("You");
writeChEnd.close();
```

**Compile command:**
`chpl foo.chpl`

**Execution command:**
`./foo`

**Associated Future Test(s):**
[`test/library/standard/IO/openwriterEndBug.chpl`](https://github.com/chapel-lang/chapel/blob/main/test/library/standard/IO/openwriterEndBug.chpl) #20451

### Configuration Information

- Output of `chpl --version`: chpl version 1.28.0 pre-release
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
CHPL_TARGET_PLATFORM: darwin
CHPL_TARGET_COMPILER: clang
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
CHPL_GMP: bundled
CHPL_HWLOC: bundled
CHPL_RE2: bundled
CHPL_LLVM: none *
CHPL_AUX_FILESYS: none
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

### PR DESCRIPTION
**Pull request by lydia-duncan**
Removes the deprecation warnings for start/end arguments and the iohint
argument.  All of these warnings were added last release and have been
cluttering the file.  We want to make a change to the region argument behavior,
to simplify that change remove the old versions instead of leaving them lying
around.

Resolves #20449 
Resolves #20450 

Impacts:
- IOHINT_NONE
- IOHINT_RANDOM
- IOHINT_SEQUENTIAL
- IOHINT_CACHED
- IOHINT_PARALLEL
- iohints
- open
- openfp
- openfd
- opentmp
- channel.seek
- openreader
- openwriter
- file.reader
- file.lines
- file.writer

Removes the old tests of these deprecation messages.  Removes the tests
of the openwriter bugs, since those were resolved by deprecating the
arguments.  Adds a .notest file to the two submitted revcomp shootouts,
since that directory's code should be kept in line with the site, but removes
the deprecation messages from the .good files and from the .perfkeys
output validation

Passed a full paratest with futures

### CODE REVIEW

File: test/studies/shootout/submitted/revcomp2.good
Comment (jeremiah-corrado) [inline test/studies/shootout/submitted/revcomp2.good:1]: This tests `.perfkeys` line number also needs to be updated to reflect the missing deprecation message. (not important now because of the notest, but I thought I'd point it out anyway).

File: test/studies/shootout/submitted/revcomp3.good
Comment (jeremiah-corrado) [inline test/studies/shootout/submitted/revcomp3.good:1]: Same thing as previous comment

**Review by jeremiah-corrado — APPROVED** [review]
Looks good to me! I just have one comment about the revcomp tests.

File: test/studies/shootout/submitted/revcomp2.good
Comment (lydia-duncan) [reply test/studies/shootout/submitted/revcomp2.good:1]: Ooooh, good catch.  Thanks!

**Comment by lydia-duncan** [comment]
The conflict is due to Ben's merge, but it looks like Michael's channel PR already does the update to make the line numbers more robust.  So I'll wait until that's merged to avoid making more trouble for Michael and resolve the merge conflicts after that

**Pull request merged**
